src/effects/VST/VSTEffect.cpp

```diff
@@ -1979,17 +1979,21 @@ bool VSTEffect::LoadParameters(
    return storeOK;
 }
 
+bool VSTEffectWrapper::SupportsChunk() const
+{
+   return (mAEffect->flags & effFlagsProgramChunks);
+}
+
 bool VSTEffect::SaveParameters(
    const RegistryPath & group, const EffectSettings &settings) const
 {
-   SetConfig(*this, PluginSettings::Private, group, wxT("UniqueID"),
-      mAEffect->uniqueID);
-   SetConfig(*this, PluginSettings::Private, group, wxT("Version"),
-      mAEffect->version);
-   SetConfig(*this, PluginSettings::Private, group, wxT("Elements"),
-      mAEffect->numParams);
+   const auto& vstSettings = GetSettings(settings);
 
-   if (mAEffect->flags & effFlagsProgramChunks)
+   SetConfig(*this, PluginSettings::Private, group, wxT("UniqueID"), vstSettings.mUniqueID );
+   SetConfig(*this, PluginSettings::Private, group, wxT("Version"),  vstSettings.mVersion  );
+   SetConfig(*this, PluginSettings::Private, group, wxT("Elements"), vstSettings.mNumParams);
```

Comment:
In the commit that adds these lines, "Rewrite SaveParameters not using the handles" -- you store values from vstSettings and not from mAeffect, which is what I meant.  Hiding all mention of mAeffect in this function isn't what I meant.  It's not wrong to factor out `SupportsChunk`, but not what I meant.

But the last commit in the sequence (after this) should also, as I said, do `StoreSettings` first, anyway, which includes the check that the values in the two agree.

Reply:
If it is allowed to reference mAEffect for that case, then I am happy to revert `SupportsChunk`